### PR TITLE
metadata: use url package to form metadata URLs

### PR DESCRIPTION
**Pull request by dorileo**
By using the url package to form metadata URLs we avoid having the weirdly formed URLs i.e. "v1//?" ...

### CODE REVIEW

File: metadata/metadata.go [hunk 1]
```diff
@@ -386,35 +390,34 @@ func (c *Client) WriteGuestAttributes(ctx context.Context, key, value string) er
...
 	if cfg.hang {
-		urlTokens = append(urlTokens, "wait_for_change=true")
-		urlTokens = append(urlTokens, "last_etag="+c.etag)
+		values.Add("wait_for_change", "true")
+		values.Add("last_etag", c.etag)
 	}
 
 	if cfg.timeout > 0 {
-		urlTokens = append(urlTokens, fmt.Sprintf("timeout_sec=%d", cfg.timeout))
+		values.Add("timeout_sec", fmt.Sprintf("%d", cfg.timeout))
```
Comment (ChaitanyaKulkarni28) [inline metadata/metadata.go:406]: isn't `timeout` already an int?

File: metadata/metadata.go [hunk 2]
```diff
@@ -318,13 +319,9 @@ func (c *Client) retry(ctx context.Context, cfg requestConfig) (string, error) {
...
-	// url.JoinPath first exists in go 1.19
-	key = strings.TrimPrefix(key, "/")
-	var reqURL string
-	if strings.HasSuffix(c.metadataURL, "/") {
-		reqURL = c.metadataURL + key
-	} else {
-		reqURL = c.metadataURL + "/" + key
+	reqURL, err := url.JoinPath(c.metadataURL, key)
+	if err != nil {
+		return "", fmt.Errorf("Failed to form metadata url: %+v", err)
```
Comment (ChaitanyaKulkarni28) [inline metadata/metadata.go:324]: nit: error strings should not be capitalized. Here and below error return messages

File: metadata/metadata.go (hunk 1 above)
Comment (dorileo) [reply metadata/metadata.go:406]: Yep, it is. `values.Add()` is `values.Add(string, string)` that's why I'm converting it to string.

File: metadata/metadata.go (hunk 2 above)
Comment (dorileo) [reply metadata/metadata.go]: Done. Thanks.

**Pull request merged**